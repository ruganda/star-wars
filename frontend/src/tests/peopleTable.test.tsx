
import TestRenderer from 'react-test-renderer';
import { MockedProvider  } from '@apollo/client/testing';
import { peopleMocks } from './mock'
import PeopleContextProvider from '../contexts/peopleContext';

import PeopleTable from '../components/peopleTable';



describe('PeopleTable', () => {

it('renders the table component without crashing', () => {
  const component = TestRenderer.create(
    <MockedProvider mocks={peopleMocks} addTypename={false}>
      <PeopleContextProvider>
        <PeopleTable data={[]} />
      </PeopleContextProvider>
    </MockedProvider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();  
});



})