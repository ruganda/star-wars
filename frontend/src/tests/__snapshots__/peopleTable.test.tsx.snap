// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleTable renders the table component without crashing 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root makeStyles-rootContainer-2 MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="a dense table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Gender
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Height
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Mass
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Homeworld
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;
